.github/workflows/dluhc-build-and-publish.yml

```
@@ -0,0 +1,181 @@
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Build dependencies
+        run: yarn build:dependencies
+
+      - name: Lint
+        run: yarn ${{matrix.app}} lint
+
+      - name: Test
+        run: yarn ${{matrix.app}} test-cov
+
+      - name: Upload test results artifacts
+        uses: actions/upload-artifact@v2
+        if: ${{ success() || failure() }}
+        with:
+          name: test-results-${{matrix.app}}
+          path: ${{matrix.app}}/test-results
+          retention-days: 14
+
+      - name: Upload test coverage artifacts
+        uses: actions/upload-artifact@v2
+        if: ${{ success() || failure() }}
+        with:
+          name: test-coverage-${{matrix.app}}
+          path: ${{matrix.app}}/test-coverage
+          retention-days: 14
+
+  setup-env-vars:
+    needs: lint-and-test
+    runs-on: ubuntu-latest
+    outputs:
+      BRANCH_NAME: ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+    steps:
+      - id: set-vars
+        run: |
+          echo "::set-output name=LOWER_CASE_BRANCH::$(
+            echo ${GITHUB_REF#refs/heads/} |
+            tr "[:upper:]" "[:lower:]"
+          )"
+      - name: echo-vars
+        id: echo-vars
+        run: |
+          echo ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+          echo $VERSION
+
+  build-and-push-images:
+    needs: [lint-and-test, setup-env-vars]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        app: [designer, runner]
+    name: build-and-push-${{matrix.app}}-image
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+      - name: Docker compose pull
+        run: docker-compose pull
+      - name: Create .env for ${{ matrix.app }} workspace
+        run: |
+          touch ./${{ matrix.app }}/.env
+          echo LAST_TAG_GH=$VERSION >> ./${{ matrix.app }}/.env
+          echo LAST_COMMIT_GH=${{ github.sha }} >> ./${{ matrix.app }}/.env
+          cat ./${{ matrix.app }}/.env
+      - name: Docker compose build ${{ matrix.app }}
+        run: |
+          LAST_TAG='$VERSION'
+          LAST_COMMIT='${{ github.sha }}'
+          docker-compose build ${{ matrix.app }}
+      - name: Log in to registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
+
+      - id: tag-and-push
+        name: Tag and push (all branches)
+        run: |
+          docker image tag ${IMAGE_NAME_STUB}${{ matrix.app }} $IMAGE_REPO_PATH/${IMAGE_NAME_STUB}${{ matrix.app }}:${{ github.sha }}
+          docker image tag ${IMAGE_NAME_STUB}${{ matrix.app }} $IMAGE_REPO_PATH/${IMAGE_NAME_STUB}${{ matrix.app }}:${{needs.setup-env-vars.outputs.BRANCH_NAME}}
+
+      - id: tag-latest
+        name: Tag Latest (if main)
+        if: needs.setup-env-vars.outputs.BRANCH_NAME == 'main'
```

```suggestion
        if: github.ref == 'refs/heads/main'
```
Think we could do this a bit more simply (as we do on other repos)

---

Agree - done